health-sdk/sdk/src/main/java/net/gini/android/health/sdk/moreinformation/MoreInformationViewModel.kt

```diff
@@ -21,27 +23,25 @@ internal class MoreInformationViewModel(private val paymentComponent: PaymentCom
         _paymentProviderAppsListFlow.asStateFlow()
 
     fun start() {
-        viewModelScope.launch {
-            if (paymentComponent == null) {
-                LOG.warn("Cannot show payment provider apps: PaymentComponent must be set before showing the BankSelectionBottomSheet")
-                return@launch
-            }
-            LOG.debug("Collecting payment provider apps state and selected payment provider app from PaymentComponent")
-            paymentComponent.paymentProviderAppsFlow.collect { paymentProviderAppsState ->
-                when (paymentProviderAppsState) {
-                    is PaymentProviderAppsState.Error -> { processError(paymentProviderAppsState.throwable) }
-                    PaymentProviderAppsState.Loading -> {}
-                    is PaymentProviderAppsState.Success -> {
-                        if (paymentProviderAppsState.paymentProviderApps.isEmpty()) {
-                            LOG.debug("No payment provider apps received")
-                        } else {
-                            LOG.debug("Received {} payment provider apps", paymentProviderAppsState.paymentProviderApps.size)
-                        }
-                        _paymentProviderAppsListFlow.value = PaymentProviderAppsListState.Success(paymentProviderAppsState.paymentProviderApps)
+        if (paymentComponent == null) {
+            LOG.warn("Cannot show payment provider apps: PaymentComponent must be set before showing the BankSelectionBottomSheet")
+            return
+        }
+        LOG.debug("Collecting payment provider apps state and selected payment provider app from PaymentComponent")
+        paymentComponent.paymentProviderAppsFlow.onEach { paymentProviderAppsState ->
+            when (paymentProviderAppsState) {
+                is PaymentProviderAppsState.Error -> { processError(paymentProviderAppsState.throwable) }
+                PaymentProviderAppsState.Loading -> {}
+                is PaymentProviderAppsState.Success -> {
+                    if (paymentProviderAppsState.paymentProviderApps.isEmpty()) {
+                        LOG.debug("No payment provider apps received")
+                    } else {
+                        LOG.debug("Received {} payment provider apps", paymentProviderAppsState.paymentProviderApps.size)
                     }
+                    _paymentProviderAppsListFlow.value = PaymentProviderAppsListState.Success(paymentProviderAppsState.paymentProviderApps)
                 }
             }
-        }
+        }.launchIn(viewModelScope)
```

Comment:
[Clarification] Can you help me understand why this change is needed? Thank you!

Reply:
[Request] I investigated a bit and found that the test failure was caused by mocking. Actually the lack of providing a value for the `paymentProviderAppsFlow` on the mocked `paymentComponent` in the `MoreInformationTest`.

According to the documentation `launchIn(scope)` is a shorthand for `scope.launch { flow.collect { } }` so this shouldn't have fixed the test failure. Maybe the `onEach()` call prevented the `kotlin.KotlinNothingValueException` exception.

Anyway simply adding a value for the flow in the `setup()` function of `MoreInformationTest.kt` solved the test failure without having to change the `MoreInformationViewModel`:
```
(...)
class MoreInformationTest {
(...)
@Before
fun setup() {
    paymentComponent = mockk(relaxed = true)
    // Provide a default empty flow to prevent kotlin.KotlinNothingValueException exceptions
    every { paymentComponent!!.paymentProviderAppsFlow } returns MutableStateFlow<PaymentProviderAppsState>(mockk()).asStateFlow()
}
(...)
```

Reply:
Oh, I just saw your comment. Now I know why `onEach` fixed. So if `onEach` wraps `paymentComponent.paymentProviderFlow` than we can do that explicitly in the test ourselves by providing an empty flow for `paymentComponent.paymentProviderFlow` by default in the test setup method.